tests/lib/forms-schema.test.js

```diff
@@ -3618,6 +3618,92 @@ describe('invalid property removal', () => {
       },
       elementSchema
     )
-    expect(error.message).toBe('child "searchUrl" fails because ["Search URL" is required]')
+    expect(error.message).toBe(
+      'child "searchUrl" fails because ["Search URL" is required]'
+    )
   })
 })
+
+test.only('should allow restrictFileTypes and restrictedFileTypes properties for File element', () => {
+  const result = Joi.validate(
+    {
+      id: 'ff9b04c3-f2ad-4994-a525-e7189eb67a79',
+      name: 'files',
+      label: 'Files',
+      type: 'file',
+      required: false,
+      restrictFileTypes: false
+    },
+    elementSchema
+  )
+  expect(result.error).toBe(null)
+
+  const { error, value } = Joi.validate(
+    {
+      id: 'ff9b04c3-f2ad-4994-a525-e7189eb67a79',
+      name: 'files',
+      label: 'Files',
+      type: 'file',
+      required: false,
+      restrictFileTypes: true,
+      restrictedFileTypes: ['png', 'jpg', 'gif']
+    },
+    elementSchema
+  )
+  expect(error).toBeFalsy()
+  expect(value).toEqual({
+    id: 'ff9b04c3-f2ad-4994-a525-e7189eb67a79',
+    name: 'files',
+    label: 'Files',
+    type: 'file',
+    readOnly: false,
+    conditionallyShow: false,
+    required: false,
+    restrictFileTypes: true,
+    restrictedFileTypes: ['png', 'jpg', 'gif']
+  })
+})
+
+test.only('should strip restrictedFileTypes if restrictFileTypes is false', () => {
+  const { error, value } = Joi.validate(
+    {
+      id: 'ff9b04c3-f2ad-4994-a525-e7189eb67a79',
+      name: 'files',
+      label: 'Files',
+      type: 'file',
+      required: false,
+      restrictFileTypes: false,
+      restrictedFileTypes: ['png']
+    },
+    elementSchema
+  )
+  expect(error).toBeFalsy()
+  expect(value).toEqual({
+    id: 'ff9b04c3-f2ad-4994-a525-e7189eb67a79',
+    name: 'files',
+    label: 'Files',
+    type: 'file',
+    readOnly: false,
+    conditionallyShow: false,
+    required: false,
+    restrictFileTypes: false
+  })
+})
+
+test.only('should only allow strings in restrictedFileTypes', () => {
```

Comment:
is it worth testing for `{restrictFileTypes: true, restrictedFileTypes: null }` ??